apps/beeai-server/src/beeai_server/utils/managed_server_client.py

```diff
@@ -37,6 +39,21 @@ class ManagedServerParameters(BaseModel):
     endpoint: str = "/sse"
 
 
+def _kill_process(process: anyio.abc.Process):
+    import psutil
+
+    parent = psutil.Process(process.pid)
+    for child in parent.children(recursive=True):  # or parent.children() for recursive=False
+        with suppress(psutil.NoSuchProcess):
+            child.kill()
+            child.wait()
+    with suppress(psutil.NoSuchProcess):
```

Comment:
Are we guaranteed to kill all the children by this point? `parent.children()` returns a list, so the result is up-to-date only at the start of iteration, new processes could be spawned in the tree during that time. Not sure how (or whether) to fix this, since there are always timing issues with this approach.